libusb/os/windows_winusb.h

```diff
@@ -45,9 +45,13 @@
 // http://msdn.microsoft.com/en-us/library/ff545978.aspx
 // http://msdn.microsoft.com/en-us/library/ff545972.aspx
 // http://msdn.microsoft.com/en-us/library/ff545982.aspx
+// These GUIDs are provided by usbiodef.h from the Windows SDK, but use of that header by non-msvc
+// compilers would result in the symbols being public, possibly causing a name collision during link.
```

Comment:
Hmm isn't it rather "the usbiodef.h in MinGW declares them as public symbols, which may clash with symbols in other objects during linking"?

But does it actually work to mix and match objects from different compilers/SDKs anyway? I thought this was not recommended and the reason for us shipping release binary libraries and objects built with different compilers. But my all means, if this mixing works, we should try keeping it working.

Reply:
Never mind, I guess you now found out that it won't be public on MinGW either.

Reply:
I think I've caused a lot of confusion here because I didn't realize mingw supported `__declspec(selectany)`, however it is supported. Additionally, even though mingw provides it's own version of the headers which define `DEFINE_GUID` and those 3 GUID values, it is correct - so my original version of this PR is OK and won't cause breaks on mingw.